packages/components/charts/src/use-radial-chart.ts

```diff
@@ -0,0 +1,238 @@
+   */
+  fillOpacity?: number | [number, number]
+}
+
+type UseRadialChartProps = UseRadialChartOptions & {
+  styles: Dict<CSSUIObject>
+}
+
+export const useRadialChart = ({
+  data: dataProp,
+  dataKey = "value",
+  innerRadius = "10%",
+  outerRadius = "80%",
+  startAngle = 180,
+  endAngle = 0,
+  fillOpacity = 1,
+  styles,
+  ...rest
+}: UseRadialChartProps) => {
+  const { theme } = useTheme()
+  const [highlightedArea, setHighlightedArea] = useState<string | null>(null)
+  const shouldHighlight = highlightedArea !== null
+  const {
+    background: backgroundProps = {},
+    dimRadialBar: dimRadialBarProps = {},
+    ...computedRadialBarProps
+  } = rest.radialBarProps ?? {}
+
+  const radialVars: CSSUIProps["var"] = useMemo(
+    () =>
+      dataProp.map(({ color }, index) => ({
+        __prefix: "ui",
+        name: `radial-bar-${index}`,
+        token: "colors",
+        value: color ?? "transparent",
+      })),
+    [dataProp],
+  )
+
+  const dimRadialBarClassName = useMemo(() => {
+    const resolvedDimRadialBar = {
+      fillOpacity: 0.3,
+      strokeOpacity: 0,
+      ...dimRadialBarProps,
+    }
+
+    return getClassName(resolvedDimRadialBar)(theme)
+  }, [dimRadialBarProps, theme])
+
+  const [chartProps, chartClassName] = useMemo(
+    () =>
+      getComponentProps<Dict, string>(
+        [rest.chartProps ?? {}, radialChartProperties],
+        styles.chart,
+      )(theme),
+    [rest.chartProps, styles.chart, theme],
+  )
+
+  const background = useMemo(
+    () =>
+      getComponentProps<Dict, string>(
+        [backgroundProps, radialBarProperties],
+        styles.background,
+      )(theme, true),
+    [backgroundProps, styles.background, theme],
+  )
+
+  const [radialBarProps, radialBarClassName] = useMemo(
+    () =>
+      getComponentProps<Dict, string>(
+        [computedRadialBarProps ?? {}, radialBarProperties],
+        styles.radialBar,
+      )(theme),
+    [computedRadialBarProps, styles.radialBar, theme],
+  )
+
+  const data = useMemo(
+    () =>
+      dataProp.map((props, index) => {
+        const {
+          name,
+          value,
+          color,
+          dimRadialBar = {},
+          ...computedProps
+        } = props
+        const dimmed = shouldHighlight && highlightedArea !== name
+        const resolvedProps = {
+          ...computedProps,
+          ...(dimmed ? dimRadialBar : { fillOpacity }),
+        }
+
+        const className = getClassName(
+          resolvedProps,
+          dimmed ? dimRadialBarClassName : undefined,
+        )(theme)
+
+        return {
+          className,
+          fill: `var(--ui-radial-bar-${index})`,
+          name,
+          value,
+          color,
+        }
+      }),
+    [
+      dataProp,
+      dimRadialBarClassName,
+      fillOpacity,
+      highlightedArea,
+      shouldHighlight,
+      theme,
+    ],
+  )
+
+  const getRadialChartProps: ChartPropGetter<
+    "div",
+    ComponentPropsWithRef<typeof Recharts.RadialBarChart>,
```

Comment:
Change `ComponentPropsWithRef<typeof Recharts.RadialBarChart>` to `ComponentPropsWithoutRef<typeof Recharts.RadialBarChart>`.